hollow/src/main/java/com/netflix/hollow/api/consumer/HollowConsumer.java

```diff
@@ -405,15 +409,16 @@ public long getToVersion() {
      */
     public static interface AnnouncementWatcher {
 
-        public static final long NO_ANNOUNCEMENT_AVAILABLE = Long.MIN_VALUE;
+        public static final long NO_VERSION_AVAILABLE = Long.MIN_VALUE;
 
         /**
-         * Return the latest announced version.
-         *
-         * @return
+         * Returns the latest announced version.
          */
         public long getLatestVersion();
 
+
+        public boolean hasPinVersion();
```

Comment:
I wonder if the code is telling us that we should be exposing an `Announcement` or `AnnouncementMetadata` class?

We would keep `getLatestVersion()` for the simple case where you only cared the version, but add a `getLatestAnnouncement()` that returned a richer object. Elsewhere, at least for the *listener* and *metrics* type interfaces we'd include the richer object instead of just `long version`. Then whether or not the version is pinned would just be a facet of information about that announcement.

Reply:
One example, we might include a key-value pair of the announced version that was overridden by the pinned version.

Reply:
I think this is a great idea, that way you could know the pinned version and the last version that was applied before the pinning. I don't know if it would be useful or not, but having a metric on how much time you spent on pinning could be another point to introduce a `AnnouncementMetadata` class.